src/admin/resolvers/queries/OpenGraphFields.ts

```diff
@@ -0,0 +1,17 @@
+import { OpenGraphFields } from '../../../shared/types';
+import getMetaData from 'metadata-scraper';
+
+/**
+ *
+ * @param parent
+ * @param data Data contains a URL of the page to request
+ */
+export async function getOpenGraphFields(
+  parent,
+  { url }
+): Promise<OpenGraphFields> {
+  const pageMetadata = await getMetaData(url, { maxRedirects: 2, timeout: 5 });
+  return {
+    description: pageMetadata?.description || null,
+  };
+}
```

Comment:
I think the return type conflicts here a little. According to the function return type we always return a `OpenGraphFields` type. In the types file, that type has the `description` property as required (`string`) but we can potentially return it as `null` here. 

I say we change the function return type to `Promise<OpenGraphFields | null>`. And have our `return` return that object or a `null`.

This will also align with our gql schema above since you have the `description` property on `OpenGraphFields` type optional as well. What do you think 🤔